From a625b708d90a04e787ce93cde9e60c67c4bc7108 Mon Sep 17 00:00:00 2001
From: Shaun Ruffell <sruffell@digium.com>
Date: Tue, 7 Dec 2010 15:57:49 -0600
Subject: [PATCH] wctdm24xxp, wcte12xp: Disable PCI read-line multiple command.

(draft)

There are some changes since DAHDI 2.2.0.2 where the read-line multiple
transaction causes packets to be dropped in the voicebus pipeline.  The
only observable behavior is that packets just go "missing" in the
pipeline.  This also only appears to affect PCI cards, and then we
haven't seen this on Hx8 PCI cards either.

This may be a root cause/related to all the IRQ misses without latency
bump issues (DAHDI-510, DAHDI-774).

Signed-off-by: Shaun Ruffell <sruffell@digium.com>
Signed-off-by: Russ Meyerriecks <rmeyerriecks@digium.com>

Index: dahdi-linux-2.4.0/drivers/dahdi/voicebus/voicebus.c
===================================================================
--- dahdi-linux-2.4.0.orig/drivers/dahdi/voicebus/voicebus.c	2010-12-24 15:10:35.426549901 +0100
+++ dahdi-linux-2.4.0/drivers/dahdi/voicebus/voicebus.c	2010-12-24 15:11:25.241548370 +0100
@@ -649,7 +649,7 @@
 	unsigned long timeout;
 	u32 reg;
 	u32 pci_access;
-	const u32 DEFAULT_PCI_ACCESS = 0xfffc0002;
+   const u32 DEFAULT_PCI_ACCESS = 0xffdc0002;
 	u8 cache_line_size;
 	BUG_ON(in_interrupt());
 
